jenkins/setup.sh

```diff
@@ -6,20 +6,41 @@ export TOP_DIR=$(dirname $(dirname $(readlink -e "${BASH_SOURCE[0]}")))
 
 export OS="$(uname)"
 
+
+compilers=(
+    cc
+    c++
+    gcc
+    g++
+    x86_64-linux-gnu-gcc
+)
+
 # setup ccache
 if [[ "$OS" == "Darwin" ]]; then
     export PATH="/usr/local/opt/ccache/libexec:$PATH"
 else
-    if [[ -d "/usr/lib/ccache" ]]; then
-        export PATH="/usr/lib/ccache:$PATH"
-    elif hash ccache > /dev/null; then
-        mkdir -p "$TOP_DIR/ccache"
-        ln -sf "$(which ccache)" "$TOP_DIR/ccache/cc"
-        ln -sf "$(which ccache)" "$TOP_DIR/ccache/c++"
-        ln -sf "$(which ccache)" "$TOP_DIR/ccache/gcc"
-        ln -sf "$(which ccache)" "$TOP_DIR/ccache/g++"
-        ln -sf "$(which ccache)" "$TOP_DIR/ccache/x86_64-linux-gnu-gcc"
-        export PATH="$TOP_DIR/ccache:$PATH"
+    if [[ -n "${SCCACHE_BUCKET}" ]]; then
+        if hash sccache 2>/dev/null; then
+            echo "SCCACHE_BUCKET is set but sccache executable is not found"
+            exit 1
+        fi
+        SCCACHE_BIN_DIR="$TOP_DIR/sccache"
+        mkdir -p "$SCCACHE_BIN_DIR"
+        for compiler in "${compilers[@]}"; do
+            ln -sf "$(which sccache)" "$SCCACHE_BIN_DIR/$compiler"
```

Comment:
The symlink didn't work in my experience. See caffe2/.jenkins/build.sh for my solution.

Reply:
@pietern I saw that and was wondering why?